pkg/provider/azure.go

```diff
@@ -303,15 +303,11 @@ func (azure *AzureProvider) Flush() error {
 }
 
 func (az *AzureProvider) Decommision(node *v1.Node) error {
-	ctx := context.Background()
-	vms := compute.NewVirtualMachinesClient(utils.ToString(az.ctx["SubscriptionId"]))
-	fut, err := vms.Delete(ctx, az.Project(), node.Name, to.BoolPtr(true))
+	kube, err := kubernetes.Kubernetes()
 	if err != nil {
-		return pluralerr.ErrorWrap(err, "failed to call deletion api")
+		return err
 	}
-
-	err = fut.WaitForCompletionRef(ctx, vms.Client)
-	return pluralerr.ErrorWrap(err, "vm deletion failed with")
+	return kube.DeleteNode(node)
```

Comment:
Does this deprovision the vm?

Reply:
good point, l will check this

Reply:
@michaeljguarino unfortunately, the VMs are still there. I will check it again

Reply:
yea i'm pretty sure we need to do what we were doing previously and kill the vm directly and allow azure's node groups to drain it automatically

Reply:
@michaeljguarino  Finally, I found a way to delete VM. We use `VirtualMachineScaleSets` for nodes. I was experimenting with the client and I found out that we need to deallocate the instance to scale down the set (otherwise will be recreated after deletion) and then we can delete the VM instance.